src/environmentd/src/lib.rs

```diff
@@ -330,7 +331,7 @@ pub async fn serve(config: Config) -> Result<Server, anyhow::Error> {
             let http_server = http::Server::new(http::Config {
                 tls: http_tls,
                 frontegg: config.frontegg,
-                adapter_client,
+                adapter_client: external_adapter_client,
```

Comment:
Where does the internal http server get its internal adapter client?

Reply:
This is a bit perplexing to me. The internal http port uses the `http::InternalServer` struct (initialized above all of this on line 204) which doesn't use an adapter client and has the following endpoints:
- `/metrics`
- `/api/livez`
- `/api/opentelemetry/config`

The external http port uses the `http::Server` struct which uses an adapter client and has the following endpoints:
- `/`
- `/api/internal/catalog` (This one is really weird since it contains "internal" in it's path but is only accessible from the external port).
- `/api/sql`
- `/memory`
- `/hierarchical-memory`
- `/prof/`
- `/static/*path`

Reply:
Oh, good point! We should move that internal catalog API to the internal HTTP server.